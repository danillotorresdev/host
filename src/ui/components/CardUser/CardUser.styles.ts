import { styled } from "@linaria/react";

export const Container = styled.div`
  display: flex;
  border-radius: 0.5rem;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  width: 100%;
  overflow: hidden;
  position: relative;
  flex-direction: column;

  @media (min-width: 768px) {
    flex-direction: row;
  }

  @media (min-width: 1024px) {
    width: 100%;
  }
`;

export const ProfileWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 100%;

  @media (min-width: 768px) {
    width: 100%;
  }

  @media (min-width: 1024px) {
    width: 40%;
  }
`;

export const Name = styled.h2`
  color: #5b5b5b;
  font-size: 1rem;
  font-size: 1.5rem;
`;

export const ParagraphWrapper = styled.div`
  display: flex;
  align-items: center;
  color: #5b5b5b;
  & > p {
    margin-left: 0.5rem;
  }
`;

export const TextBox = styled.div`
  padding: 0 1rem;
  display: flex;
  flex-direction: column;
  width: 100%;

  @media (min-width: 768px) {
    width: 60%;
  }
`;

export const ProfileImageWrapper = styled.div`
  height: 400px;
  width: 100%;
`;
