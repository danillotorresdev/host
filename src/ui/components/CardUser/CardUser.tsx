import { Icon } from "@/ui/components/Icon/Icon";
import * as S from "./CardUser.styles";
import { LazyBackgroundImg } from "@/ui/components/LazyBackgroundImage/LazyBackgroundImage";
import { LinkButton } from "@/ui/components/LinkButton/LinkButton";

export type CardUserProps = {
  name: string;
  followers: number;
  following: number;
  email: string | null;
  bio: string;
  avatar_url: string;
  username: string;
};

export const CardUser = ({
  name,
  followers,
  following,
  email,
  bio,
  avatar_url,
  username,
}: CardUserProps) => (
  <S.Container>
    <S.ProfileWrapper>
      <S.ProfileImageWrapper>
        <LazyBackgroundImg img={avatar_url} />
      </S.ProfileImageWrapper>
    </S.ProfileWrapper>
    <S.TextBox>
      <S.Name>{name}</S.Name>
      <S.ParagraphWrapper>
        <Icon name="followers" color="#5b5b5b" size="md" />
        <p>Seguidores: {followers}</p>
      </S.ParagraphWrapper>
      <S.ParagraphWrapper>
        <Icon name="followers" color="#5b5b5b" size="md" />
        <p>Seguindo: {following}</p>
      </S.ParagraphWrapper>
      <S.ParagraphWrapper>
        <Icon name="email" color="#5b5b5b" size="md" />
        {email ? <p>{email}</p> : <p>Email não disponível</p>}
      </S.ParagraphWrapper>
      <S.ParagraphWrapper>
        <p>{bio}</p>
      </S.ParagraphWrapper>
      <div
        style={{
          width: "200px",
          marginTop: "1rem",
        }}
      >
        <LinkButton
          to={`/user/${username}/repos`}
          style={{
            marginBottom: "1rem",
          }}
        >
          <Icon name="arrow-foward" color="#b1aeae" /> Repositórios
        </LinkButton>
      </div>
    </S.TextBox>
  </S.Container>
);
