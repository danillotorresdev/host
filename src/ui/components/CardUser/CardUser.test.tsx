
import { CardUser } from './CardUser';
import { render, screen } from "@/utils/settings/tests/utilities";
import { expect, it, describe } from "vitest";
import { BrowserRouter } from 'react-router-dom';

describe("CardUser", () => {
  it("should render the user's name", () => {
    const { asFragment } = render(
      <CardUser
        name="John Doe"
        followers={100}
        following={200}
        email="jhondoe@email.com"
        bio="Lorem ipsum dolor sit amet"
        avatar_url="https://avatars.githubusercontent.com/u/1?v=4"
        username="jhondoe"
      />,
      {
        wrapper: BrowserRouter,
      }
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText("John Doe")).toBeDefined();
  }
  )
});