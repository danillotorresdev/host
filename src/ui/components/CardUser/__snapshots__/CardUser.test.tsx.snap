// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardUser > should render the user's name 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0 c1eelvmx"
  >
    <div
      class="mocked-styled-1 p1n234za"
    >
      <div
        class="mocked-styled-5 p1bhgu3m"
      >
        <div
          style="background-image: url(https://avatars.githubusercontent.com/u/1?v=4); background-repeat: no-repeat; background-position: center center; background-size: cover; filter: blur(20px); transition: filter 0.5s; width: 100%; height: 100%;"
        >
          <img
            alt=""
            src="https://avatars.githubusercontent.com/u/1?v=4"
            style="display: none;"
          />
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-4 tlp2jul"
    >
      <h2
        class="mocked-styled-2 ngisml4"
      >
        John Doe
      </h2>
      <div
        class="mocked-styled-3 pxniugy"
      >
        <svg
          class="svg svg-followers"
          data-testid="icon-svg"
          style="width: 24px; height: 24px; color: rgb(91, 91, 91);"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="/assets/sprite.svg#followers"
          />
        </svg>
        <p>
          Seguidores: 100
        </p>
      </div>
      <div
        class="mocked-styled-3 pxniugy"
      >
        <svg
          class="svg svg-followers"
          data-testid="icon-svg"
          style="width: 24px; height: 24px; color: rgb(91, 91, 91);"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="/assets/sprite.svg#followers"
          />
        </svg>
        <p>
          Seguindo: 200
        </p>
      </div>
      <div
        class="mocked-styled-3 pxniugy"
      >
        <svg
          class="svg svg-email"
          data-testid="icon-svg"
          style="width: 24px; height: 24px; color: rgb(91, 91, 91);"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="/assets/sprite.svg#email"
          />
        </svg>
        <p>
          jhondoe@email.com
        </p>
      </div>
      <div
        class="mocked-styled-3 pxniugy"
      >
        <p>
          Lorem ipsum dolor sit amet
        </p>
      </div>
      <div
        style="width: 200px; margin-top: 1rem;"
      >
        <a
          class="mocked-styled-6 ltiwt9s"
          href="/user/jhondoe/repos"
          style="margin-bottom: 1rem;"
        >
          <svg
            class="svg svg-arrow-foward"
            data-testid="icon-svg"
            style="width: 24px; height: 24px; color: rgb(177, 174, 174);"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="/assets/sprite.svg#arrow-foward"
            />
          </svg>
           Reposit√≥rios
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
